lib/avromatic.rb

```diff
@@ -77,19 +77,38 @@ def self.prepare!(skip_clear: false)
         end
       end
     end
-    eager_load_models!
+    eager_load_models!(skip_nested_models: skip_clear)
   end
 
   def self.eager_load_models=(models)
     @eager_load_model_names = Array(models).map { |model| model.is_a?(Class) ? model.name : model }
   end
 
-  def self.eager_load_models!
+  def self.eager_load_models!(skip_nested_models: true)
     (@eager_load_model_names || []).each do |model_name|
-      nested_models.ensure_registered_model(model_name.constantize)
+      model = model_name.constantize
+      next if !nested_models.ensure_registered_model(model) || skip_nested_models
```

Comment:
Relying on `ensure_registered_model` to return a truthy value is a bit dangerous given the current implementation (i.e. it works today but likely was not intentional). Adding a `next if nested_models.registered?(model)` guard is safer.